### PR TITLE
Revert the latest integrate to fix issues with `--iree-codegen-llvmgpu-enable-transform-dialect-jit`

### PR DESCRIPTION
**Pull request by kuhar**
This reverts the latest integrate an a post-integrate workaround:
* Revert "Integrate llvm-project at 523640f7a4c0 (#12419)" 908c427d0b724853894acd06fed2cb04835269b7
* Revert "Work around test failures in `mobilenetv3_fake_weights` (#12445)" 51bdf0a9c58b1374f605f7aec3495919f94b2890

Issue: https://github.com/openxla/iree/issues/12444

### CODE REVIEW

**Review by GMNGeoffrey — APPROVED** [review]
Thank you. I know it's painful to revert these, but agreed that we really shouldn't be disabling core tests

**Comment by kuhar** [comment]
I'll wait for at least one more approval before landing

**Review by stellaraccident — APPROVED** [review]

**Pull request merged**
